packages/engine/Source/Renderer/Context.js

```diff
@@ -31,183 +30,43 @@ import TextureCache from "./TextureCache.js";
-  }
-
-  const glWrapper = {};
-
-  // JavaScript linters normally demand that a for..in loop must directly contain an if,
-  // but in our loop below, we actually intend to iterate all properties, including
-  // those in the prototype.
-  /*eslint-disable guard-for-in*/
-  for (const propertyName in gl) {
-    const property = gl[propertyName];
-
-    // wrap any functions we encounter, otherwise just copy the property to the wrapper.
-    if (property instanceof Function) {
-      glWrapper[propertyName] = wrapFunction(property);
-    } else {
-      Object.defineProperty(
-        glWrapper,
-        propertyName,
-        makeGetterSetter(gl, propertyName, logFunction)
-      );
-    }
-  }
-  /*eslint-enable guard-for-in*/
-
-  return glWrapper;
-}
-
-function getExtension(gl, names) {
-  const length = names.length;
-  for (let i = 0; i < length; ++i) {
-    const extension = gl.getExtension(names[i]);
-    if (extension) {
-      return extension;
-    }
-  }
-
-  return undefined;
-}
-
 /**
  * @private
  * @constructor
+ *
+ * @param {HTMLCanvasElement} canvas The canvas element to which the context will be associated
+ * @param {ContextOptions} [options] Options to control WebGL settings for the context
  */
 function Context(canvas, options) {
-  // this check must use typeof, not defined, because defined doesn't work with undeclared variables.
-  if (typeof WebGLRenderingContext === "undefined") {
-    throw new RuntimeError(
-      "The browser does not support WebGL.  Visit http://get.webgl.org."
-    );
-  }
-
   //>>includeStart('debug', pragmas.debug);
   Check.defined("canvas", canvas);
   //>>includeEnd('debug');
 
-  this._canvas = canvas;
-
-  options = clone(options, true);
-  // Don't use defaultValue.EMPTY_OBJECT here because the options object gets modified in the next line.
-  options = defaultValue(options, {});
-  options.allowTextureFilterAnisotropic = defaultValue(
-    options.allowTextureFilterAnisotropic,
-    true
-  );
-  const webglOptions = defaultValue(options.webgl, {});
+  const {
+    getWebGLStub,
+    requestWebgl2 = false,
+    webgl: webglOptions = {},
+    allowTextureFilterAnisotropic = true,
+  } = defaultValue(options, {});
 
   // Override select WebGL defaults
   webglOptions.alpha = defaultValue(webglOptions.alpha, false); // WebGL default is true
   webglOptions.stencil = defaultValue(webglOptions.stencil, true); // WebGL default is false
+  webglOptions.powerPreference = defaultValue(
+    webglOptions.powerPreference,
+    "high-performance"
+  ); // WebGL default is "default"
 
-  const requestWebgl2 =
-    defaultValue(options.requestWebgl2, false) &&
-    typeof WebGL2RenderingContext !== "undefined";
-  let webgl2 = false;
-
-  let glContext;
-  const getWebGLStub = options.getWebGLStub;
-
-  if (!defined(getWebGLStub)) {
-    if (requestWebgl2) {
-      glContext =
-        canvas.getContext("webgl2", webglOptions) ||
-        canvas.getContext("experimental-webgl2", webglOptions) ||
```

Comment:
@lilleyse Can you confirm if we can safely get rid of `experimental-webgl2`?

According to [WebGL2fundemantals.org](https://webgl2fundamentals.org/webgl/lessons/webgl1-to-webgl2.html):
![image](https://user-images.githubusercontent.com/5172619/199788705-c60bb55f-c6c2-4d58-b5f9-8c912fbdb844.png)

According to [CanIUse.com](https://caniuse.com/webgl2): 
![image](https://user-images.githubusercontent.com/5172619/199789065-1af3cf5f-a559-49fd-93f5-24ecd600acb8.png)

Reply:
I don't have a definitive answer, but seems safe to remove.